projectgenerator/libqgsprojectgen/dataobjects/project.py

```diff
@@ -95,8 +95,7 @@ def create(self, path: str, qgis_project: QgsProject):
 
         qgis_project.setCrs(self.crs)
 
-        if self.legend:
-            self.legend.create(qgis_project)
+        self.legend.create(qgis_project)
```

Comment:
Removing this might break some cases where a legend is not manually specified or generated.

Remember that the idea is that this should also be usable without ili2pg or pg involved